lintcheck/src/main.rs

```diff
@@ -90,16 +90,71 @@ struct Crate {
     options: Option<Vec<String>>,
 }
 
+/// A single emitted output from clippy being executed on a crate. It may either be a
+/// `ClippyWarning`, or a `RustcIce` caused by a panic within clippy. A crate may have many
+/// `ClippyWarning`s but a maximum of one `RustcIce` (at which point clippy halts execution).
+#[derive(Debug)]
+enum ClippyCheckOutput {
+    ClippyWarning(ClippyWarning),
+    RustcIce(RustcIce),
+}
+
+#[derive(Debug)]
+struct RustcIce {
+    pub crate_name: String,
+    pub message: String,
+    pub backtrace: String,
+}
+impl RustcIce {
+    pub fn from_stderr(crate_name: &str, stderr: &str) -> Option<Self> {
+        if stderr.contains("panicked at") {
+            let lines = stderr.split('\n');
+
+            let mut reading_message = false;
+            let mut message_lines = vec![];
+
+            let mut reading_backtrace = false;
+            let mut backtrace_lines = vec![];
+
+            for line in lines {
+                if line.contains("panicked at") {
+                    reading_message = true;
+                    message_lines.push(line.to_owned());
+                } else if line.contains("stack backtrace:") {
+                    reading_message = false;
+                    reading_backtrace = true;
+                    backtrace_lines.push(line.to_owned());
+                } else if line.contains("the compiler unexpectedly panicked") {
```

Comment:
you can probably steal some from here: :) 
https://github.com/matthiaskrgr/icemaker/blob/master/src/main.rs#L1810